### PR TITLE
Pmail V1

### PR DESCRIPTION
**Pull request by mcdaniel67**

### CODE REVIEW

File: pmail/inbox.html [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    data-toggle="collapse"
+                    data-target="#bs-example-navbar-collapse-1"
+                    aria-expanded="false">
+                <span class="sr-only">Toggle navigation</span>
+                <span class="icon-bar"></span>
+                <span class="icon-bar"></span>
+                <span class="icon-bar"></span>
+            </button>
+            <a class="navbar-brand" href="#">
+                <span class="glyphicon glyphicon-envelope"></span> Pmail</a>
```
Comment (danvayn) [inline pmail/inbox.html]: Make sure your markups consistent :D

File: pmail/inbox.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            Mail <span class="caret"></span>
+        </button>
+        <ul class="dropdown-menu">
+            <li><a href="#">Mail</a></li>
+            <li><a href="#">Contacts</a></li>
+            <li><a href="#">Tasks</a></li>
+        </ul>
+        <button type="button" class="btn btn-default btn-sm"
+                id="check_all"
+                data-toggle="button" style="margin-left: 20%">
```
Comment (danvayn) [inline pmail/inbox.html]: inline styles are a nono!

File: pmail/inbox.html [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            Move to Folder
+            <span class="caret"></span>
+        </button>
+        <ul class="dropdown-menu">
+            <li><a href="#">User Folder 1 <span class="badge">19</span></a></li>
+            <li><a href="#">User Folder 2 <span class="badge">84</span></a></li>
+            <li><a href="#">New Folder</a></li>
+        </ul>
+    </div>
+    <div class="btn-group col-md-1">
```
Comment (danvayn) [inline pmail/inbox.html]: col-md-1 is a dangerous thing..on lower widths the button doesnt show itself fully

File: pmail/inbox.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
+example_emails();
+handle_checks();
+
+// Obviously we want to delete this when the backend exists.
+function example_emails() {
+    generate_message(false, 'John Smith', 'Lorem Ipsum', '12-12-12', 'label-warning', 'Important');
```
Comment (danvayn) [inline pmail/inbox.js]: generate message probably should be able to accept an array of objects so you don't have to copy paste a method 10 times over

File: pmail/inbox.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    generate_message(true, 'Shae Loveall', 'Lorem Ipsum', '10-12-12');
+    generate_message(true, 'Rico Heyman', '*don\'t open this email*', '10-12-12');
+    generate_message(true, 'Nico Heypan', 'What Can You Afford?', '09-12-12');
+    generate_message(true, 'Tantum Guest', 'Lorem Ipsum', '08-12-12', 'label-warning', 'Important');
+    generate_message(true, 'Joann Ha', 'What are our customers saying?', '08-10-12');
+    generate_message(true, 'John Doe', 'Where to Drink Beer Right Now', '08-03-12');
+}
+
+// Used to toggle visible folders when we click more/less
+function show_folders() {
```
Comment (danvayn) [inline pmail/inbox.js]: btw these methods are polluting the global namescape, check how you can avoid that

File: pmail/inbox.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    return Text
+        .toLowerCase()
+        .replace(/ /g, '')
+        .replace(/[^\w-]+/g, '')
+        ;
+}
+
+// Select all emails
+function select_all() {
+    var select_all = !$(document).find('#check_all').hasClass('active');
```
Comment (danvayn) [inline pmail/inbox.js]: i wouldnt give this variable the same name as the method itself, and also i would do $("#check_all.active").length to check if its checked

File: pmail/inbox.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    $('input:checkbox[id^="check_"]').each(function () {
+        var read = $(this).closest('.email').hasClass('read') ? '#F0F0F0' : '#FFFFFF';
+        var color = select_all ? '#D6D6D6' : read;
+        $(this).prop('checked', select_all);
+        $(this).closest('.email').css('background', color);
+    });
+}
+
+// Redirects the browser using email information. (Ludicrously unsafe)
```
Comment (danvayn) [inline pmail/inbox.js]: LOL

File: pmail/inbox.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        "&subject=" + to_slug(subject) +
+        "&date=" + to_slug(date);
+}
+
+// Handle the checkboxes and stars for each email.
+function handle_checks() {
+    $('#check_all').click(select_all);
+    number_emails = $('.email').size();
+    for (var i = 1; i <= number_emails; i++) {
+        handle_message(i);
```
Comment (danvayn) [inline pmail/inbox.js]: i think theres a better way to do this..

File: pmail/inbox.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    }
+}
+
+// Handle a particular message. (highlighting, clicking, etc)
+function handle_message(index) {
+    $('.sender_subject_' + index)
+        .click(function () {
+            email_click_redirect(this);
```
Comment (danvayn) [inline pmail/inbox.js]: yeah theres definitely a better way to do this

File: pmail/inbox.js [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+/* Appends a message to the listgroup ul.
+ * \params
+ * Label must be passed as a bootstrap label.
+ * Check out the examples for more details how the labels work...
+ * read is a boolean corresponding to whether or not the message has been read.
+ * The rest of the parameters are rather self explanatory
+ */
+function generate_message(read, author, subject, date, label, label_message) {
```
Comment (danvayn) [inline pmail/inbox.js]: :mask:

File: pmail/js/bootstrap.js [hunk 11]
```diff
@@ -0,0 +1,2363 @@
...
+
+
+  // ALERT DATA-API
+  // ==============
+
+  $(document).on('click.bs.alert.data-api', dismiss, Alert.prototype.close)
+
+}(jQuery);
+
+/* ========================================================================
```
Comment (danvayn) [inline pmail/js/bootstrap.js]: id put all the bootstrap junk in its own file and just include it into your markup

File: pmail/inbox.css [hunk 12]
```diff
@@ -0,0 +1,139 @@
+* {
```
Comment (danvayn) [inline pmail/inbox.css]: use sass instead of css!!!!!

File: pmail/inbox.css [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+    position: fixed;
+}
+
+.folders {
+    padding-top: 10%;
+    text-align: left;
+}
+
+.folders li {
+    padding-left: 10%;
```
Comment (danvayn) [inline pmail/inbox.css]: using percentages to add padding is dangerous because your elements can end up being TOO responsive

**Comment by djkotowski** [comment]
Nice work!  Take a look at the comments and refactor where needed (particularly the inline styles).  No resubmission needed.

**Pull request closed without merging**